### DISCUSSION TITLE
Help users when detail view shortcode does not have ID

### DISCUSSION THREAD

**Discussion by jmaas-onoffice** [onOffice-Web-Org/oo-wp-plugin#331] [_Discarded pitches]
# Users don't understand the error message
Currently, when users want to preview the detail page, they are presented with this error message:
![grafik](https://user-images.githubusercontent.com/90763364/190999375-956a018f-ad53-45ec-ba0f-d56904a1bed4.png)

We often receive questions what to do about this, because users cannot know that they need to go to the estate list and choose an estate there to open the URL with a valid ID.

# Give instructions on what to do
Instead of showing the error message, we should tell users that they need to go to an estate list and open an estate detail page from there.

If the user is logged in, we should additionally allow them to easily open a random estate's detail page. This should not be available publicy, only when the user is logged into WordPress. The estates can be any accessible estates, since they all are published for the website. Therefore we should for example give a link to a random estate's detail page.

**Comment by dai-eastgate** [comment]
@jmaas-onoffice Can you tell me the UI details of the message?

**Reply by jmaas-onoffice** [reply]
Could you do a quick placeholder with Lorem Ipsum at first and send a screenshot here? I'll work out a message in the meantime. :)

**Reply by dai-eastgate** [reply]
![image](https://user-images.githubusercontent.com/106214469/192712365-82bca1e9-239a-47e3-aefa-76bf8e351496.png)

**Reply by jmaas-onoffice** [reply]
Message:

> You have opened the detail page, but we do not know which estate to show you, because there is no estate ID in the URL. Please go to an estate list and open an estate from there.

For admin users, below that and above the link:

> Since you are logged in, here is a link to a random estate so that you can preview the detail page:

And how about we change the button to be a normal link with the estate's title as the text? So for example:

> Since you are logged in, here is a link to a random estate so that you can preview the detail page:
> [Beautiful home with great view](#)

**Reply by dai-eastgate** [reply]
Can you explain more specifically about requirement " For admin users, below that and above the link:
Since you are logged in, here is a link to a random estate so that you can preview the detail page:"
Is it only for admin role or all roles are logged?

**Reply by jmaas-onoffice** [reply]
Good question!

I think all roles that are logged into WordPress is fine. The important thing is that it is not shown to regular visitors.
